tests/Controller/FoldersControllerTest.php

```diff
@@ -22,70 +22,78 @@
 namespace OCA\Mail\Tests\Controller;
 
 use Horde_Imap_Client_Exception;
+use Horde_Imap_Client_Socket;
+use OCA\Mail\Account;
+use OCA\Mail\Contracts\IMailManager;
 use OCA\Mail\Controller\FoldersController;
+use OCA\Mail\Folder;
+use OCA\Mail\Service\AccountService;
 use OCP\AppFramework\Db\DoesNotExistException;
 use OCP\AppFramework\Http;
 use OCP\AppFramework\Http\JSONResponse;
-use PHPUnit_Framework_TestCase;
+use OCP\IRequest;
+use PHPUnit_Framework_MockObject_MockObject;
+use Test\TestCase;
 
-class FoldersControllerTest extends PHPUnit_Framework_TestCase {
+class FoldersControllerTest extends TestCase {
 
-	private $controller;
+	/** @var string */
 	private $appName = 'mail';
+
+	/** @var IRequest|PHPUnit_Framework_MockObject_MockObject */
 	private $request;
+
+	/** @var AccountService|PHPUnit_Framework_MockObject_MockObject */
 	private $accountService;
+
+	/** @var string */
 	private $userId = 'john';
 
-	public function setUp() {
-		$this->request = $this->getMockBuilder('OCP\IRequest')
-			->getMock();
-		$this->accountService = $this->getMockBuilder('OCA\Mail\Service\AccountService')
-			->disableOriginalConstructor()
-			->getMock();
-		$this->controller = new FoldersController($this->appName, $this->request,
-			$this->accountService, $this->userId);
-	}
+	/** @var IMailManager|PHPUnit_Framework_MockObject_MockObject */
+	private $mailManager;
 
-	public function folderDataProvider() {
-		$files = [
-			'folders_german',
-		];
-		// Add directory prefix to tests/data
-		$data = array_map(function ($file) {
-			$path = dirname(__FILE__) . '/../data/' . $file . '.json';
-			return [json_decode(file_get_contents($path), true)];
-		}, $files);
-
-		// Add empty account = no folders
-		array_push($data, [
-				[
-				'folders' => [
-				],
-			],
-		]);
+	/** @var FoldersController */
+	private $controller;
 
-		return $data;
+	public function setUp() {
+		$this->request = $this->createMock(IRequest::class);
+		$this->accountService = $this->createMock(AccountService::class);
+		$this->mailManager = $this->createMock(IMailManager::class);
+		$this->controller = new FoldersController($this->appName, $this->request, $this->accountService, $this->userId, $this->mailManager);
 	}
 
-	/**
-	 * @dataProvider folderDataProvider
-	 */
-	public function testIndex($data) {
-		$account = $this->getMockBuilder('OCA\Mail\Account')
-			->disableOriginalConstructor()
-			->getMock();
+	public function testIndex() {
```

Comment:
````
/**
 * Some documented method, does foo things and need bar for its purpose. You have to enter a better description than this
 *
 * @param string $bar Some bar string
 * @return void
 */
public function doFoo (string $bar) {
    // Do foo things with bar
[...]
}
````
That is a so called doc-tag. :-) It will produce nice documentation in such tools (see `Doxygen`, it can process more than just C/C++).